### PR TITLE
K95 Platinum preliminary support (dirty)

### PR DESCRIPTION
**Pull request by Kuro-Maii**
added a test to see if the devices is a K95 Platinum specifically
if yes make a fake init packet and the keyboard will magical work

Known issue:
the set firmware is in the wrong byte order with in this fix
if the order is made correct the deamon will not work with the K95 Platinum like it does without this "fix"
i.e. BROKEN PIPE

### CODE REVIEW

**Comment by ghost** [comment]
Very nice!

Please, reopen a new PR and recommit the changes with properly set email address so that everyone can identify the committer later.

Can you also describe what features are not working or what features do work? If I get it right, now the daemon learned to detect that the keyboard is indeed K95 Platinum, did you test anything further?

File: src/ckb-daemon/usb.h [hunk 1]
```diff
@@ -39,6 +39,7 @@
 #define P_K95_PLATINUM  0x1b2d
 #define P_K95_PLATINUM_STR "1b2d"
 #define IS_K95(kb)      ((kb)->vendor == V_CORSAIR && ((kb)->product == P_K95 || (kb)->product == P_K95_NRGB || (kb)->product == P_K95_PLATINUM))
+#define IS_K95P(kb)     ((kb)->vendor == V_CORSAIR && ((kb)->product == P_K95_PLATINUM))
```
Comment (ghost) [inline src/ckb-daemon/usb.h:42]: If there's a test specifically for K95 Platinum now, is there a need to keep testing for it in IS_K95 macro?

**Review by ghost — CHANGES_REQUESTED** [review]
Broken authorship.

**Comment by tatokis** [comment]
I can understand exactly where you're coming from, and that you want to get your keyboard working, however I am sorry to say, but I do not want this getting merged/making it into release.

People are free to use it until we get a proper solution, but I do not want ugly hacks like this.

Instead, we really should figure out what is going wrong, and only if we can't, after a lot of attempts, resort to ugly workarounds.

**Comment by ghost** [comment]
Also, please try following https://chris.beams.io/posts/git-commit/ when writing commit messages for everyone to make clear what's happened in a long-term perspective.

File: src/ckb-daemon/usb.h (hunk 1 above)
Comment (tatokis) [reply src/ckb-daemon/usb.h:42]: Yes, it's still a K95, the P is used to only apply the workarounds for the specific device. I do think it should be called IS_PLATINUM though, as I sense more devices coming up.

Comment (Kuro-Maii) [reply src/ckb-daemon/usb.h:42]: shall i make it IS_PLATINUM or IS_K95_PLATINUM?
I can do this before commiting my work again and opening a new pull request

**Comment by Kuro-Maii** [comment]
I agree completely with the fact that it is a dirty hack.
and I do wan't my keyboard to work, that is true.
however, as was discovered by reading the packet stream between CUE and the keyboard there was NO response from the keyboard to the init packed, or so I was told in the IRC channel.

That said, I do wan't a better solution and want to keep working on it till i find it or move the hack to a more clean function... ( suggestions are more then welcome )
I am also considering other users who might want to use there K95 Platinum under linux and are now stubbing their toe on the not working driver.
And for them a dirty hack may work the best till the better solution has been implemented.

**Comment by tatokis** [comment]
Closing this, as a solution without dirty hacks was found and reported working.
(See #53 for a description.)

Thanks to everyone involved!

**Pull request closed without merging**